app/src/main/java/com/kleshchin/danil/filemanager/MainActivity.java

```diff
@@ -0,0 +1,118 @@
+package com.kleshchin.danil.filemanager;
+
+import android.content.Intent;
+import android.graphics.PorterDuff;
+import android.graphics.drawable.Drawable;
+import android.net.Uri;
+import android.os.Bundle;
+import android.os.Parcelable;
+import android.support.v7.app.AppCompatActivity;
+import android.support.v7.widget.Toolbar;
+import android.view.MenuItem;
+import android.view.View;
+import android.webkit.MimeTypeMap;
+import android.widget.AdapterView;
+import android.widget.EditText;
+import android.widget.ListView;
+
+import java.io.File;
+
+public class MainActivity extends AppCompatActivity {
+    private ListView listView;
+    private ListAdapter listAdapter;
+    private String mainPath = "/storage";
+    private String currentDirectory;
+    private EditText toolbarTitle;
+    public static final String LISTVIEW_STATE = "listview.state";
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_main);
+        listView = (ListView) findViewById(R.id.listView);
+        toolbarTitle = (EditText) findViewById(R.id.toolbar_title);
+        initToolbar();
+        fillListView(new File(mainPath));
+        listView.setOnItemClickListener(new AdapterView.OnItemClickListener() {
+            @Override
+            public void onItemClick(AdapterView<?> adapterView, View view, int i, long l) {
+                String path = toolbarTitle.getText().toString() + "/" + listAdapter.getItem(i).getName();
+                if (new File(listAdapter.getItem(i).getPath()).isDirectory()) {
+                    fillListView(new File(listAdapter.getItem(i).getPath()));
+                } else {
+                    onFileSelected(listAdapter.getItem(i).getPath());
+                }
+            }
+        });
+    }
+
+    private void onFileSelected(String path) {
```

Comment:
Can be easily converted into a static method. Of course, it should have another name in this case.